ui/Main.elm

```diff
@@ -0,0 +1,341 @@
+    Decode.map5 Release
+        (Decode.field "builtins" <| Decode.list decodeSystemAddon)
+        (Decode.field "updates" <| Decode.maybe <| Decode.list decodeSystemAddon)
+        (Decode.field "release" decodeReleaseDetails)
+        (Decode.field "id" Decode.string)
+        (Decode.field "last_modified" Decode.int)
+
+
+client : Kinto.Client
+client =
+    Kinto.client
+        "https://kinto-ota.dev.mozaws.net/v1/"
+        (Kinto.Basic "user" "pass")
+
+
+recordResource : Kinto.Resource Release
+recordResource =
+    Kinto.recordResource "systemaddons" "versions" decodeRelease
+
+
+getReleaseList : Cmd Msg
+getReleaseList =
+    client
+        |> Kinto.getList recordResource
+        |> Kinto.sortBy [ "-release.version" ]
+        |> Kinto.send ReleasesFetched
+
+
+init : ( Model, Cmd Msg )
+init =
+    ( { releases = []
+      , filteredReleases = []
+      , channels = []
+      , filters =
+            { channels = []
+            }
+      }
+    , getReleaseList
+    )
+
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Sub.none
+
+
+viewReleaseDetails : ReleaseDetails -> Html.Html Msg
+viewReleaseDetails details =
+    Html.div []
+        [ Html.dl []
+            [ Html.dt [] [ Html.text "URL" ]
+            , Html.dd [] [ Html.text details.url ]
+            ]
+        , Html.dl []
+            [ Html.dt [] [ Html.text "Build ID" ]
+            , Html.dd [] [ Html.text details.buildId ]
+            ]
+        , Html.dl []
+            [ Html.dt [] [ Html.text "Target" ]
+            , Html.dd [] [ Html.text details.target ]
+            ]
+        , Html.dl []
+            [ Html.dt [] [ Html.text "Lang" ]
+            , Html.dd [] [ Html.text details.lang ]
+            ]
+        , Html.dl []
+            [ Html.dt [] [ Html.text "Channel" ]
+            , Html.dd [] [ Html.text details.channel ]
+            ]
+        ]
+
+
+joinBuiltinsUpdates : List SystemAddon -> Maybe (List SystemAddon) -> List SystemAddonVersions
+joinBuiltinsUpdates builtins updates =
+    builtinVersions builtins
+        |> updateVersions updates
+        |> List.sortBy .id
+
+
+builtinVersions : List SystemAddon -> List SystemAddonVersions
+builtinVersions builtins =
+    List.map (\a -> SystemAddonVersions a.id (Just a.version) Nothing) builtins
+
+
+updateVersions : Maybe (List SystemAddon) -> List SystemAddonVersions -> List SystemAddonVersions
+updateVersions updates builtins =
+    let
+        uupdates =
+            Maybe.withDefault [] updates
+
+        mergeVersion : SystemAddonVersions -> SystemAddon -> SystemAddonVersions
+        mergeVersion b u =
+            SystemAddonVersions b.id b.builtin (Just u.version)
+
+        hasBuiltin : List SystemAddonVersions -> SystemAddon -> Bool
+        hasBuiltin addons addon =
+            List.any (\i -> i.id == addon.id) addons
+
+        addToBuiltin : List SystemAddonVersions -> SystemAddon -> List SystemAddonVersions
+        addToBuiltin addons addon =
+            List.map
```

Comment:
> More generally, having id as a field inside a record (rather than outside it) is a code smell for me.

You should enjoy Kinto a lot then (/me runs away)